tests/conftest.py

```diff
@@ -0,0 +1,97 @@
+# sets up database for tests
+
+import pytest
+import testing.postgresql
+
+from sqlalchemy import create_engine, text
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy_sqlschema import maintain_schema
+
+# test data and formatting functions
+from sample_data import WAT1, WAT2, WAT3, BAS1, make_region
+from sample_data import CHUM, PNKO, PNKE, make_taxon
+from sample_data import CUC1, CUC2, CUPO, CUPE, make_cu, make_population
+
+# import salmon_occurrence
+from salmon_occurrence import salmon_db, Region
+
+
+# create database
+@pytest.fixture()
+def db_uri():
+    with testing.postgresql.Postgresql() as pg:
+        yield pg.url()
+
+
+# initialize database with postgis and sqlalchemy
+@pytest.fixture()
+def db_engine(db_uri):
+    engine = create_engine(db_uri)
+
+    connection = engine.connect()
+    connection.execute(text("CREATE EXTENSION postgis;"))
+    connection.execute(text("CREATE SCHEMA IF NOT EXISTS salmon_geometry;"))
+    connection.execute(text("GRANT CREATE ON SCHEMA salmon_geometry TO postgres"))
```

Comment:
`postgres` is a very privileged user. I wonder if it would be better to create a user with ordinary privilege and use it for the tests. Possibly does not matter much.